MovingBlocks/Terasology
# Fix/transient deps

This is meant to address the problem described in #4014: unresolved module dependencies when they're not all present as source.

### How to Test

See #4014 for notes on what kind of brokenness we were seeing. The basic test plan is:

* Start with nothing in `modules/` (neither sources nor jars)
* `groovyw module get JoshariasSurvival` **without** using `recurse`
* start a game and make a new JoshariasSurvival world
  * from `gradlew game`
  * from the IDE `TerasologyPC` run config

You might also test other modules or game modes with deep dependency trees.

### Outstanding before merging

- [x] this branch is based on #4342; merge it or rebase this branch. (The fix doesn't **depend** on that, but it does speed up testing a bit.)
- [x] does it work? (right now I have it working with JoshariasSurvival and CoreSampleGameplay in `modules/`, does it work with JS alone and no CSG?)
- [x] why do I have both `Workstation-1.1.0.jar` **and** `Workstation-1.1.0-SNAPSHOT.jar`?
- [x] does it mess up the distribution tasks?

#### to do later (out of scope for bugfix)
- should we do something similar with the `:moduleJars` task?
- can we get rid of RemoteModuleGatherer???
